### PR TITLE
feat(Predictions.PubSub): Include related vehicles

### PR DESCRIPTION
**Pull request by KaylaBrady**
### Summary

_Ticket:_ [Predictions Scalability: new channel that publishes predictions updates in chunks](https://app.asana.com/0/1205732265579288/1207791938443975/f)

What is this PR for?

This PR adds a new Vehicle store that is updated by a single stream for all vehicles. 
Those vehicles are fetched as part of fetching predictions for parity with the existing predictions channel.

### CODE REVIEW

File: lib/mbta_v3_api/store/vehicles.ex [hunk 1]
```diff
@@ -0,0 +1,198 @@
+defmodule MBTAV3API.Store.Vehicles do
```
Comment (EmmaSimon) [inline lib/mbta_v3_api/store/vehicles.ex:1]: This looks all identical to `Store.Predictions` except for the Store class and the Impl, would it be at all straightforward to make something more generic that could just take those two params and generate the module? Or is it possible/likely that other ones will have different implementations?

Comment (EmmaSimon) [reply lib/mbta_v3_api/store/vehicles.ex:1]: Er, well, I guess alerts are the only other realtime stream, do we plan to store those in ETS too, or just these two? If it's just going to be two then I'm more fine with the repetition, but three is pushing it.

Comment (KaylaBrady) [reply lib/mbta_v3_api/store/vehicles.ex:1]: Yeah that's fair! I don't have plans to convert alerts as part of this ticket (but there will be a bunch of follow-up tasks, like converting the vehicles channel to use the new vehicles table & predictions by trip to use the new trips table). 

I did a time boxed pass of making this more generic in a few ways:
* macro to hide the boilerplate calling the correct module for testing
* common timed_fetch function

Predictions & Vehicles are still mostly similar, but Predictions has the extra logic for pulling in other related records. I think any further refactoring to make these common would be worth holding off on.

**Comment by KaylaBrady** [comment]
Where the vehicle stream is subscribed to needs to move - will push that in a bit!

File: lib/mbta_v3_api/store/vehicles.ex (hunk 1 above)
Comment (EmmaSimon) [reply lib/mbta_v3_api/store/vehicles.ex:1]: That makes sense, this looks great, thank you!

**Review by EmmaSimon — APPROVED** [review]

**Pull request merged**